api/common/crypto.cc

```diff
@@ -99,7 +99,7 @@ bool EncAes256CbcSha512(const std::string &key,
     // Build last block content with padding
     memcpy(last_block, clear_last_block, clear.length() % AES_BLOCK_SIZE);
     memset(last_block + AES_BLOCK_SIZE - pad_length, pad_length,
-           AES_BLOCK_SIZE);
+           AES_BLOCK_SIZE - clear.length() % AES_BLOCK_SIZE);
```

Comment:
you already know pad_length so I would say
`memset(last_block + AES_BLOCK_SIZE - pad_length, pad_length, pad_length);`
Check this in tests [crypto.cc unit tests](https://github.com/outscale-nts/butterfly/blob/f6f803bd83cd5b13d56fd630fb58e7ac76f55201/api/common/crypto.cc#L229)